src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunctionFactory.cs

```diff
@@ -825,6 +825,23 @@ static bool IsAsyncMethod(MethodInfo method)
                     {
                         try
                         {
+                            if (value is string potentiallyJsonString)
+                            {
+                                Debug.Assert(typeInfo.Type != typeof(string), "string parameters should not enter this branch.");
+
+                                // Account for the parameter potentially being a JSON string.
+                                // This is needed for compatibility with Semantic Kernel,
+                                // which may pass parameters as JSON strings.
+                                try
+                                {
+                                    return JsonSerializer.Deserialize(potentiallyJsonString, typeInfo);
+                                }
+                                catch (JsonException)
```

Comment:
If we think this might be common, is there any screening we can do on potentiallyJsonString to avoid the cost of the exception as much as possible?

Reply:
Added in https://github.com/dotnet/extensions/pull/6572/commits/77d324af616a4d7d2f20e0ead63af7082426d91e